llvm/tools/sycl-post-link/SpecConstants.cpp

```diff
@@ -330,9 +331,16 @@ void collectCompositeElementsDefaultValuesRecursive(
       collectCompositeElementsDefaultValuesRecursive(M, El, LocalOffset,
                                                      DefaultValues);
     }
+    const size_t SLSize = SL->getSizeInBytes();
+
+    // Additional padding may be needed at the end of the struct if size does
+    // not match the number of bytes inserted.
+    while (DefaultValues.size() < BaseDefaultValueOffset + SLSize)
```

Comment:
May we use DefaultValues.resize(BaseDefaultValueOffset + SLSize) if DefaultValues.size() is less?

Reply:
Good point! I have added this.